### ISSUE TITLE
TCP receive blocks on OSX: MavConnection::recv never returns messages

### ISSUE THREAD

**Issue by tstellanova** [mavlink/rust-mavlink#12]
Testing this library with a TCP connection, send seems to work but receive does not. 
The problem seems to be that the socket read has no timeout by default, so the read waits forever. 

Setting the timeout:
```
sock_clone.set_read_timeout(Some(Duration::from_millis(100))).unwrap();
```
and then loop-ignoring the resulting `io::ErrorKind::WouldBlock` seems to unstick the socket.  I've got a WIP fork for this -- if this repo is still maintained, I can submit a PR for that when it's ready.

**Comment by podhrmic** [comment]
@tstellanova I believe this should be fixed in #13 ?

**Comment by tstellanova** [comment]
Yes, fixed in #13

**Comment by tstellanova** [comment]
Fixed
